sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala

```diff
@@ -462,4 +463,27 @@ class OrcQuerySuite extends QueryTest with BeforeAndAfterAll with OrcTest {
       }
     }
   }
+
+  test("column nullability and comment - write and then read") {
+    val schema = StructType(
+      StructField("cl1", IntegerType, nullable = false,
+        new MetadataBuilder().putString("comment", "test").build()) ::
```

Comment:
: ) It is a little bit hacky. Maybe we should add a new API for users to add comments?